app/controllers/list_controller.rb

```diff
@@ -257,7 +257,11 @@ def top(options = nil)
     options ||= {}
     period = params[:period]
     period ||= ListController.best_period_for(current_user.try(:previous_visit_at), options[:category])
-    public_send("top_#{period}", options)
+    if ListController.action_methods.include? "top_#{period}"
```

Comment:
Is is possible to validate the `period` variable instead of checking whether a method has been defined?

Reply:
Yea, totally. I originally just went with the NoMethodFound because it wouldn't be defined if it wasn't in the period table, but yea there could be issues with that. I'll just check if its a valid period value.